File: tessoku/src/bin/A17.rs
use proconio::input;

fn main() {
    input! {
        n: usize,
        a: [usize; n - 1],
        b: [usize; n - 2],
    }

    let mut dp = vec![0; n];
    dp[0] = 0;
    dp[1] = a[0];
    for i in 2..n {
        dp[i] = (dp[i - 1] + a[i - 1]).min(dp[i - 2] + b[i - 2]);
    }

    let mut res = vec![];
    let mut curr = n;
    loop {
        res.push(curr);
        if curr == 1 {
            break;
        }

        if dp[curr - 2] + a[curr - 2] == dp[curr - 1] {
            curr -= 1;
        } else {
            curr -= 2;
        }
    }

    println!("{}", res.len());
    for v in res.iter().rev() {
        print!("{} ", v);
    }
}

File: tessoku/python/A17.py
n = int(input())
a = [0] + list(map(int, input().split()))
b = [0] * 2 + list(map(int, input().split()))

dp = [0] * n
dp[1] = a[1]
for i in range(2, n):
    one = dp[i - 1] + a[i]
    two = dp[i - 2] + b[i]
    dp[i] = min(one, two)

ans = []
curr = n - 1
while curr >= 0:
    ans.append(curr + 1)
    if dp[curr] == dp[curr - 1] + a[curr]:
        curr -= 1
    else:
        curr -= 2

print(len(ans))
print(" ".join(map(str, reversed(ans))))
